lib/secure_cookies.rb

```diff
@@ -15,17 +14,14 @@ def call(env)
     status, headers, body = @app.call(env)
 
     if (cookie_header = headers['Set-Cookie']).present?
-      cookies = cookie_header.split(COOKIE_SEPARATOR)
 
-      cookies.each do |cookie|
+      Array(cookie_header).each do |cookie|
         next if cookie.blank?
 
         cookie << '; Secure' if env['HTTPS'] == 'on' && !cookie.match?(SECURE_REGEX)
         cookie << '; HttpOnly' if !cookie.match?(HTTP_ONLY_REGEX)
         cookie << '; SameSite=Lax' if !cookie.match?(SAME_SITE_REGEX)
       end
-
-      headers['Set-Cookie'] = cookies.join(COOKIE_SEPARATOR)
```

Comment:
does this work because we're now mutating the `Set-Cookie` header directly with the `<<`'s above?

Reply:
yep